doc/config_yang_validation/config_db_yang_validation.md

```diff
@@ -0,0 +1,178 @@
+
+|  Rev  |  Date   |      Author      | Change Description |
+| :---: | :-----: | :--------------: | ------------------ |
+|  1.0  | 08/2022 |     Isabel Li    | Phase 1 Design     |
+
+## Project Overview  
+
+### Problem Statement
+Currently, SONiC config fields updates are validated ad-hoc, and much of the validation criteria are duplicated both in python (hard-coded validation specs) and in YANG models
+
+### Goal
+Refactor code to utilize preexisting YANG models for config validation, instead of duplicating the validation specs in both YANG and in python
+
+### Scope
+
+#### Python
+1.	SONiC config fields updated via config CLI (e.g. config portchannel add <…>). A comprehensive list of config fields in this project’s scope is as follows, defined in sonic-utilities/config directory:
+    - 	AAA 
+    - 	ACL 
+    - 	BGP
+    - 	Console 
+    - 	DHCP Relay 
+    - 	Drop Counter 
+    - 	Dynamic Buffer Management
+    - 	ECN
+    - 	Feature
+    - 	Interface 
+    - 	Interface Naming Mode
+    - 	Interface Vrf Binding
+    - 	IPv6 Link Local
+    - 	Kubernetes
+    - 	Linux Kernel Dump
+    - 	Loopback Interfaces
+    - 	VRF
+    - 	Management VRF
+    - 	Mirroring
+    - 	Muxcable
+    - 	NAT Enable/Disable YANG Validation
+    - 	NTP
+    - 	NVGRE
+    - 	PBH
+    - 	Platform Component Firmware
+    - 	PortChannel
+    - 	QoS
+    - 	sFlow
+    - 	SNMP
+    - 	Subinterfaces
+    - 	Syslog
+    - 	VLAN
+    - 	Warm Restart
+    - 	Watermark
+    - 	ZTP
+3.	ACL rules: incremental update, delete
+4.	pfc
+5.	crm
+6.	mclag
+7.	counterpoll 
+8.	Scripts (python)
+a.	configlet
+b.	db_migrator
+c.	dropconfig
+d.	mellanox_buffer_migrator
+e.	neighbor_advertisor
+f.	null_route_helper
+g.	hostcfgd
+
+#### C++
+1. buffermgrd
+
+
+## Case Study
+
+### Portchannel
+
+PortChannel config CLI is defined in sonic-utilities/config/main.py. In this file, we find the following config validation:  
+![](../../images/config_yang_validation/adhoc_portchannel.png)
+
+However, these specs for validation are also defined in the sonic-portchannel YANG model:  
+![](../../images/config_yang_validation/yang_portchannel.png)
+
+For PortChannel and many of the config fields listed above, the field validation is unnecessarily duplicated in sonic-utilities/config and in YANG models. The goal of this project is to utilize preexisting YANG models to validate config CLI field updates, without unnecessarily defining separate, redundant validation specs ad-hoc in sonic-utilities. 
+
+During the migration process, we will first leave the ad-hoc validation code in-place, and leave an option to configure type of validation used: ad-hoc or YANG validation. Once YANG validation has stabilized and YANG validation coverage has widened, we will remove the preexisting ad-hoc validation code for non performance-sensitive scenarios. 
+
+In performance-sensitive scenarios, such as warm and fast reboot, ad-hoc validation will remain the mode of validation, as GCU is more time-intensive.
```

Comment:
I don't think warm and fast boot will use CLI to reconfigure as the configurations are directly loaded from config_db through reconciliation. 
It would be good if you can capture the performance penalty for using this approach against adhoc validation for some sample command.

Reply:
Thank you, I will take a closer look at writes to ConfigDB for warm and fast boot and update accordingly. I don't expect CLI to be used, but I would expect to possibly find some adhoc validation written in.

Reply:
Oh yes, sorry I missed your comment @dgsudharsan . I added a note to the HLD to address the performance penalty